docs/en/api/S3-API.md

```diff
@@ -18,375 +18,1701 @@ sharing with file system based applications, and storage system abstraction (e.g
+    "Owner": null,
+    "StorageClass": "STANDARD"
+}
 ```
+{% endnavtab %}
+{% endnavtabs %}
 
-#### Abort a multipart upload
-
-A non-completed upload can be aborted:
 
+#### [UploadPartCopy](https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPartCopy.html)
+{% navtabs %}
+{% navtab AWS CLI %}
 ```console
-$ curl -i -X DELETE http://localhost:39999/api/v1/s3/testbucket/testobject?uploadId=3
-
-HTTP/1.1 204 No Content
-Date: Tue, 18 Jun 2019 21:37:27 GMT
-Server: Jetty(9.2.z-SNAPSHOT)
+$ aws --profile alluxio-s3 --endpoint "http://localhost:39999/api/v1/s3/" s3api upload-part-copy \
+  --bucket=testbucket --key=multipart_copy.txt --upload-id=117440512004 --part-number=1 --copy-source="testbucket/test.txt"
+{
+    "CopyPartResult": {
+        "ETag": "911df44b7ff57801ca8d74568e4ebfbe",
+        "LastModified": "2022-05-03T13:00:13.572000+00:00"
+    }
+}
+
+$ aws --profile alluxio-s3 --endpoint "http://localhost:39999/api/v1/s3/" s3api list-parts \
+  --bucket=testbucket --key=multipart_copy.txt --upload-id=117440512004
+{
+    "Parts": [
+        {
+            "PartNumber": 1,
+            "LastModified": "2022-05-03T13:00:13.584000+00:00",
+            "ETag": "\"\"",
+            "Size": 27040
+        }
+    ],
+    "ChecksumAlgorithm": null,
+    "Initiator": null,
+    "Owner": null,
+    "StorageClass": "STANDARD"
+}
 ```
+{% endnavtab %}
+{% navtab REST Clients %}
+```console
+$ curl -i -H "Authorization: AWS4-HMAC-SHA256 Credential=testuser/... SignedHeaders=... Signature=..." \
+  -H "x-amz-copy-source: testbucket/test.txt" \
+  -X PUT -T "${ALLUXIO_HOME}/LICENSE" "http://localhost:39999/api/v1/s3/testbucket/multipart.txt?uploadId=134268059649&partNumber=1"
+HTTP/1.1 100 Continue
+
+HTTP/1.1 200 OK
+Date: Tue, 03 May 2022 23:48:56 GMT
+Content-Type: application/xml
+Content-Length: 135
+Server: Jetty(9.4.43.v20210629)
 
-#### Delete an empty bucket
+<CopyObjectResult>
+  <LastModified>2022-05-03T16:48:56.594Z</LastModified>
+  <ETag>911df44b7ff57801ca8d74568e4ebfbe</ETag>
+</CopyObjectResult>
 
-```console
-$ curl -i -X DELETE http://localhost:39999/api/v1/s3/testbucket
+$ curl -i -H "Authorization: AWS4-HMAC-SHA256 Credential=testuser/... SignedHeaders=... Signature=..." \
+  -X GET "http://localhost:39999/api/v1/s3/testbucket/multipart.txt?uploadId=134268059649"
+HTTP/1.1 200 OK
+Date: Tue, 03 May 2022 23:49:05 GMT
+Content-Type: application/octet-stream
+Content-Length: 314
+Server: Jetty(9.4.43.v20210629)
 
-HTTP/1.1 204 No Content
-Date: Tue, 18 Jun 2019 21:38:38 GMT
-Server: Jetty(9.2.z-SNAPSHOT)
+<ListPartsResult>
+  <Bucket>/testbucket</Bucket>
+  <Key>multipart.txt</Key>
+  <UploadId>134268059649</UploadId>
+  <StorageClass>STANDARD</StorageClass>
+  <IsTruncated>false</IsTruncated>
+  <Part>
+    <PartNumber>1</PartNumber>
+    <LastModified>2022-05-03T16:48:56.602Z</LastModified>
+    <ETag>""</ETag>
+    <Size>27040</Size>
+  </Part>
+</ListPartsResult>
 ```
+{% endnavtab %}
+{% endnavtabs %}
 
 ### Python S3 Client
```

Comment:
Python S3 client examples are untouched and should be considered out-of-date (eg: they are missing examples for tagging)

Reply:
i assume the existing python example still works as described; the incoming changes should have no effect on this?